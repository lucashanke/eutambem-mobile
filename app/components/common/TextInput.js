import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextInput as NativeTextInput } from 'react-native';
import appStyles from '../../styles';
import { MAX_TEXT_INPUT_LENGTH } from '../../constants';

class TextInput extends Component {

  inputRef = (component) => { this.input = component; };

  placeholder = () => `${this.props.placeholder} ${!this.props.required ? '(Opcional)' : ''}`;

  componentDidUpdate = () => {
    if (this.props.blur) {
      this.input.blur();
    }
  };

  render() {
    let style = [appStyles.input];
    let autoGrow = false;
    if (this.props.multiline) {
      style = style.concat(appStyles.multilineInput);
      autoGrow = true;
    }

    return (
      <NativeTextInput
        {...this.props}
        style={style}
        maxLength={MAX_TEXT_INPUT_LENGTH}
        multiline={this.props.multiline}
        placeholder={this.placeholder()}
        autoGrow={autoGrow}
        ref={this.inputRef}
      />
    );
  }
}

TextInput.propTypes = {
  multiline: PropTypes.bool,
  placeholder: PropTypes.string.isRequired,
  required: PropTypes.bool,
};

TextInput.defaultProps = {
  multiline: false,
  required: false,
};

export default TextInput;
