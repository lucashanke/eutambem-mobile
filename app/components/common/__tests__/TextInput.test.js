import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { TextInput as NativeTextInput } from 'react-native';

import { TextInput } from '../TextInput';

const props = {
  placeholder: 'Test TextInput',
  onValueChange: sinon.spy(),
};

describe('TextInput', () => {
  it('renders native TextInput', () => {
    const wrapper = shallow(<TextInput {...props} />);

    expect(wrapper.find(NativeTextInput)).toHaveLength(1);
  });

  it('onChangeText of native TextInput calls onValueChange prop', () => {
    const wrapper = shallow(<TextInput {...props} />);

    wrapper
      .find(NativeTextInput)
      .props()
      .onChangeText('new text');

    expect(props.onValueChange.calledWith('new text')).toBeTruthy();
  });

  it('sets multiline and autoGrow to true in native input if multiline prop is true', () => {
    const wrapper = shallow(<TextInput
      {...props}
      multiline
    />);

    const nativeInput = wrapper.find(NativeTextInput);

    nativeInput.props().onChangeText('new text');

    expect(nativeInput.props().multiline).toBeTruthy();
    expect(nativeInput.props().autoGrow).toBeTruthy();
  });
});
