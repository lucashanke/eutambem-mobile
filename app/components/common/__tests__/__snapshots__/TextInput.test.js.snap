// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ForwardRef(View)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      multiline={false}
      onValueChange={[Function]}
      placeholder="Test TextInput"
      required={false}
    />
  </ForwardRef(View)>
</ForwardRef(View)>
`;
